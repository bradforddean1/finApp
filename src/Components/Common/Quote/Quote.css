.Quote {
	display: flex;
	flex-direction: column;
}

.Quote * {
	margin-bottom: 0.44em;
}

.Quote *:nth-last-child(0) {
	margin-bottom: 0;
}
