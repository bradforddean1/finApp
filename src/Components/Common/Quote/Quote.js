import React from "react";
import "./Quote.scss";
import PropTypes from "prop-types";

/**
 * Display sceuoruty current price and change for previous close.
 * @component
 */
function Quote(props) {
	const { exchange, current, currency, change, changePct } = props;
	return (
		<div className="Quote card-line">
			<span className="emphasis">
				{current} {currency}
			</span>
			<span className="emphasis">
				{change} ({changePct})
			</span>
			{!!exchange && <p className="de-emphasis">Data from {exchange}</p>}
		</div>
	);
}

Quote.propTypes = {
	exchange: PropTypes.string,
	current: PropTypes.string,
	currency: PropTypes.string,
	change: PropTypes.string,
	changePct: PropTypes.string,
};

Quote.defaultProps = {
	exchange: null,
	current: "N/A",
	currency: "",
	change: "N/A",
	changePct: "N/A",
};

export default Quote;
