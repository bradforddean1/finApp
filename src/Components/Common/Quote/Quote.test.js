import React from 'react';
import ReactDOM from 'react-dom';
import Quote from './Quote';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('Quote Component', () => {
    it('Renders',
        () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <Quote />,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it.skip('Renders the Default State',
    () => {
        const wrapper = shallow(<Quote />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it.skip('Renders given props',
    () => {
        const wrapper = shallow(<Quote {...props}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
