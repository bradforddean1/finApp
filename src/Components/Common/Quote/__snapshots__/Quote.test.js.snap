// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Component Renders given props 1`] = `
<div
  className="Quote card-line"
>
  <span
    className="emphasis"
  >
    $3.22
     
    USD
  </span>
  <span
    className="emphasis"
  >
    -1.2
     (
    -.02%
    )
  </span>
  <p
    className="de-emphasis"
  >
    Data from 
    NYSE
  </p>
</div>
`;

exports[`Quote Component Renders the Default State 1`] = `
<div
  className="Quote card-line"
>
  <span
    className="emphasis"
  >
    N/A
     
  </span>
  <span
    className="emphasis"
  >
    N/A
     (
    N/A
    )
  </span>
</div>
`;
